app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php

```diff
@@ -14,6 +16,23 @@ class Delete extends \Magento\Backend\App\Action
      * @see _isAllowed()
      */
     const ADMIN_RESOURCE = 'Magento_Cms::page_delete';
+    
+    /**
+     * @var \Magento\Cms\Api\PageRepositoryInterface
+     */
+    private $cmsPage;
+    
+    /**
+     * @param Context $context
+     * @param \Magento\Cms\Api\PageRepositoryInterface $cmsPage
+     */
+    public function __construct(
+        Context $context,
+        \Magento\Cms\Api\PageRepositoryInterface $cmsPage
```

Comment:
Please make new dependency optional (you already added fallback to object manager) due to backward compatibility reasons

Reply:
Done.